flow/project.py

```diff
@@ -1712,28 +1725,29 @@ def scheduler_jobs(self, scheduler):
         for sjob in self._expand_bundled_jobs(scheduler.jobs()):
             yield sjob
 
-    def _get_operations_status(self, job, cached_status):
-        "Return a dict with information about job-operations for this job."
+    def _get_operations_status(self, jobs, cached_status):
+        "Return a dict with information about job-operations for this aggregate."
         starting_dict = functools.partial(dict, scheduler_status=JobStatus.unknown)
         status_dict = defaultdict(starting_dict)
         for group in self._groups.values():
-            completed = group._complete((job,))
-            eligible = False if completed else group._eligible((job,))
-            scheduler_status = cached_status.get(group._generate_id((job,)),
-                                                 JobStatus.unknown)
-            for operation in group.operations:
-                if scheduler_status >= status_dict[operation]['scheduler_status']:
-                    status_dict[operation] = {
-                        'scheduler_status': scheduler_status,
-                        'eligible': eligible,
-                        'completed': completed
-                    }
-
-        for key in sorted(status_dict):
-            yield key, status_dict[key]
+            if jobs in self._get_aggregates(group.name):
+                completed = group._complete(jobs)
+                eligible = not completed and group._eligible(jobs)
+                scheduler_status = cached_status.get(group._generate_id(jobs),
+                                                     JobStatus.unknown)
+                for operation in group.operations:
+                    if scheduler_status >= status_dict[operation]['scheduler_status']:
+                        status_dict[operation] = {
+                            'scheduler_status': scheduler_status,
+                            'eligible': eligible,
+                            'completed': completed
+                        }
```

Comment:
Just because a `FlowGroup` is eligible does not imply all its operations are.

Reply:
Good catch. This should be fixed and we should add a test for the correctness of this behavior.

Reply:
I think to check the eligibility of a `FlowGroup` we iterate on all the operations in that group and then decide whether a group is eligible or not. This means that all the operations have to eligible in order for the group to be eligible.

Reply:
@kidrahahjo you are right that we check the operations within a group to determine if the group itself is eligible, we only require that one operation in a group to be eligible for the entire group to be considered eligible. For multi-operation groups this means that not every operation within a group will be eligible.

Reply:
This is because a multi-operation group can have eligibility dependencies that are resolved _during_ execution of the group. For example, a group of (`initialize`, `simulate`, `analyze`) can be submitted, where only `initialize` is eligible at first, and `simulate` depends on `initialize` and `analyze` depends on `simulate`.

Reply:
Interesting point, I didn't realize this.
I will address the changes.

Reply:
@b-butler @bdice I had one question, what if a particular operation was included in multiple groups?
Wouldn't the information get overwritten? 

I may be missing some very basic point here but I'm unable to figure out how should we proceed.

Reply:
I'm going to defer to @b-butler on this, I'm not sure.

Reply:
The status output we show for jobs doesn't include the "actual" flow groups.
I believe this problem (if it's actually a problem) is already in our code.

Reply:
If it is overwritten, it shouldn't change the correctness. It is sub-optimal though. Ideally we would separate singleton groups from the rest and get the status of those operations.

Reply:
I'll go with this approach and re-request a review.